// @cloudinary/url-gen/actions/effect.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var effect = require('../effect-9ab58868.cjs');
require('../Blur-bfbb28a9.cjs');
require('../Qualifier-6633a22f.cjs');
require('../QualifierValue-e770d619.cjs');
require('../QualifierModel-0923d819.cjs');
require('../unsupportedError-74070138.cjs');
require('../Action-0ed405c1.cjs');
require('../FlagQualifier-7b069f22.cjs');
require('../region-ef039f04.cjs');
require('../NamedRegion-38e9cc4d.cjs');
require('../RectangleRegion-2672a877.cjs');
require('../Accelerate-269acc42.cjs');
require('../Loop-20af27a6.cjs');
require('../LeveledEffectAction-c6a11f05.cjs');
require('../SimpleEffectAction-aadd7e24.cjs');
require('../internalConstants-6e675c29.cjs');
require('../Cartoonify-10d02441.cjs');
require('../Outline-c98a41df.cjs');
require('../prepareColor-c03e99eb.cjs');
require('../MakeTransparent-468ef2ac.cjs');
require('../Vectorize-83429cdb.cjs');
require('../SimulateColorBlind-2a1253bf.cjs');
require('../EffectActionWithLevel-69a976c7.cjs');
require('../AssistColorBlind-d89f5f85.cjs');
require('../GradientFade-5fbf7b64.cjs');
require('../FadeOut-dfe0f0fc.cjs');
require('../Colorize-012fe385.cjs');
require('../Shadow-35824aa8.cjs');
require('../StyleTransfer-34e93b6e.cjs');
require('../Dither-6aebc058.cjs');
require('../Deshake-acd938db.cjs');
require('../Pixelate-f1f2f0f1.cjs');
require('../EffectActionWithStrength-32a97380.cjs');
require('../Blackwhite-888c6b63.cjs');
require('../FadeIn-db605d72.cjs');
require('../RemoveBackgroundAction-bdeeab1a.cjs');
require('../Theme-84161b18.cjs');
require('../BackgroundRemoval-8758e15c.cjs');
require('../DropShadow-76097e92.cjs');
require('../GenerativeRemove-fb3990b0.cjs');
require('../GenerativeReplace-688ba16f.cjs');
require('../GenerativeRecolor-427c8b81.cjs');
require('../GenerativeBackgroundReplace-204fa1ef.cjs');



exports.Effect = effect.Effect;
exports.accelerate = effect.accelerate;
exports.advancedRedEye = effect.advancedRedEye;
exports.artisticFilter = effect.artisticFilter;
exports.assistColorBlind = effect.assistColorBlind;
exports.backgroundRemoval = effect.backgroundRemoval;
exports.blackwhite = effect.blackwhite;
exports.blur = effect.blur;
exports.boomerang = effect.boomerang;
exports.cartoonify = effect.cartoonify;
exports.colorize = effect.colorize;
exports.deshake = effect.deshake;
exports.dither = effect.dither;
exports.dropShadow = effect.dropShadow;
exports.enhance = effect.enhance;
exports.fadeIn = effect.fadeIn;
exports.fadeOut = effect.fadeOut;
exports.generativeBackgroundReplace = effect.generativeBackgroundReplace;
exports.generativeRecolor = effect.generativeRecolor;
exports.generativeRemove = effect.generativeRemove;
exports.generativeReplace = effect.generativeReplace;
exports.generativeRestore = effect.generativeRestore;
exports.gradientFade = effect.gradientFade;
exports.grayscale = effect.grayscale;
exports.loop = effect.loop;
exports.makeTransparent = effect.makeTransparent;
exports.negate = effect.negate;
exports.noise = effect.noise;
exports.oilPaint = effect.oilPaint;
exports.outline = effect.outline;
exports.pixelate = effect.pixelate;
exports.redEye = effect.redEye;
exports.removeBackground = effect.removeBackground;
exports.reverse = effect.reverse;
exports.sepia = effect.sepia;
exports.shadow = effect.shadow;
exports.simulateColorBlind = effect.simulateColorBlind;
exports.styleTransfer = effect.styleTransfer;
exports.theme = effect.theme;
exports.transition = effect.transition;
exports.upscale = effect.upscale;
exports.vectorize = effect.vectorize;
exports.vignette = effect.vignette;
